pepper/apps/hash_transform.c

```diff
@@ -2,29 +2,15 @@
 #include "dex_common.h"
 #include "hashing.h"
 
-struct ShaHash { bool digest[256]; };
-
 void compute(struct In *input, struct Out *output){
     // Read private input
     struct Private pInput = readPrivateInput();
-    struct ShaHash shaHash[1] = { 0 };
-
-    bool shaIn[512] = { 0 };
-    uint32_t index;
-    for (index=0; index < ORDERS*253; index+=253) {
-        // sha hash
-        copyBits(shaHash->digest, 0, shaIn, 0, 256);
-        copyBits(pInput.orders, index, shaIn, 259, 253);
-        ext_gadget(shaIn, shaHash, 0);
-        uint32_t result = sumBits(shaHash->digest, 256);
-    }
-
-    // assert final sha hash is equal with public input
-    int128 resultL = sumBits(shaHash->digest, 128);
-    int128 resultR = sumBits(shaHash->digest+128, 128);
    
-    assert_zero(resultL - input->shaHashL);
-    assert_zero(resultR - input->shaHashR);
+    int128 result[2];
+    hashSHA(pInput.orders, ORDERS*253, 253, result);
```

Comment:
Just a little note: 
You might wanna change the function signatures, so that they are more similar. 

hashSHA is void and pederson hashes have a return type.

But it's not a must

Reply:
The problem with returning arrays is memory management. The memory for the contents of the array is allocated on the stack and released once we leave the function. This means if some function calls hashSHA and wants to access the returned values later, it will operate on invalid memory.
Therefore, the array has to be allocated outside the actual function call and only populated inside hashSHA.
I could make pedersenHash take a field* and assign it, but for single values variables I found it simpler to just return the value.
Alternatively, hashSHA could return a struct with two fields lResult and rResult.

Reply:
ah okay.
Then we could keep it like it is or make pedersonHash take a field* and assign it.

As you prefer. Both options are fine for me.

Reply:
👍 Changed it to return a struct. You are right, that we should be consistent if we can. Thanks for pushing 😃